src/llm/reason/prompt.py

```diff
@@ -75,7 +79,7 @@
 [JSON RESPONSE FORMAT]
 {JSON_SCHEMA_STR}
 
-Determine which next command to use, and respond using the format specified above:
+Determine which next command to use, check if in [RECENT EPISODES] or [RELATED PAST EPISODES] messages if you already did same action and args for the same task and avoid doing it again to avoid infinite loop, and respond using the format specified above:
```

Comment:
Change to:
Determine which next command to use, check in [RECENT EPISODES] and [RELATED PAST EPISODES] messages if you already performed the same action and args for the same task. Avoid doing it again to prevent infinite loops. Respond using the format specified above: